From e2bda454b21bdc93195b9bf5ee022dadd26e826b Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:02:52 -0400
Subject: [fs] btrfs: fix EINVAL checks in btrfs_clone

Message-id: <1407179109-13714-286-git-send-email-sandeen@redhat.com>
Patchwork-id: 87248
O-Subject: [RHEL7.1 PATCH 285/422] Btrfs: fix EINVAL checks in btrfs_clone
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Chris Mason <clm@fb.com>

commit 3a29bc0928003674f45b4fe625b4d0738a22c60d
Author: Chris Mason <clm@fb.com>
Date:   Mon Apr 7 07:10:40 2014 -0700

    Btrfs: fix EINVAL checks in btrfs_clone

    btrfs_drop_extents can now return -EINVAL, but only one caller
    in btrfs_clone was checking for it.  This adds it to the
    caller for inline extents, which is where we really need it.

    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index 86393b7..8802d36 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -3141,8 +3141,9 @@ process_slot:
         new_key.offset + datal,
         1);
     if (ret) {
-     btrfs_abort_transaction(trans, root,
-        ret);
+     if (ret != -EINVAL)
+      btrfs_abort_transaction(trans,
+       root, ret);
      btrfs_end_transaction(trans, root);
      goto out;
     }
-- 
1.7.1